infra/modules/aws-psoxy-rest/main.tf

```diff
@@ -8,35 +8,29 @@ terraform {
   }
 }
 
-resource "aws_lambda_function" "psoxy-instance" {
-  function_name    = var.function_name
-  role             = aws_iam_role.iam_for_lambda.arn
-  handler          = var.handler_class
-  runtime          = "java11"
-  filename         = var.path_to_function_zip
-  source_code_hash = var.function_zip_hash
-  timeout          = 55  # seconds
-  memory_size      = 512 # megabytes
-
-  environment {
-    variables = merge(var.environment_variables, yamldecode(file(var.path_to_config)))
-  }
-}
-
-# cloudwatch group per lambda function
-resource "aws_cloudwatch_log_group" "lambda-log" {
-  name              = "/aws/lambda/${aws_lambda_function.psoxy-instance.function_name}"
-  retention_in_days = 7
+module "psoxy_lambda" {
+  source = "../aws-psoxy-lambda"
+
+  function_name        = var.function_name
+  handler_class        = "co.worklytics.psoxy.Handler"
+  path_to_function_zip = var.path_to_function_zip
+  function_zip_hash    = var.function_zip_hash
+  memory_size_mb       = 512
+  timeout_seconds      = 55
+  aws_assume_role_arn  = var.aws_assume_role_arn
+  path_to_config       = var.path_to_config
+  source_kind          = var.source_kind
+  parameters           = []
 }
 
 resource "aws_lambda_function_url" "lambda_url" {
```

Comment:
This can be removed, because it is already done in `infra/modules/aws-psoxy-lambda/main.tf:32`, right?